.github/workflows/samples-tools/cmd/affected/main.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"slices"
+
+	"samples-tools/pkg/utils"
+)
+
+func main() {
+	usage := "usage: affected path/to/config.json [head-commit] [main-commit]"
+	configPath := utils.ArgRequired(1, usage)
+	headCommit := utils.ArgWithDefault(2, "HEAD")
+	mainCommit := utils.ArgWithDefault(3, "origin/main")
+
+	config, err := utils.LoadConfig(configPath)
+	if err != nil {
+		panic(err)
```

Comment:
can you add more context to the err messages so we know where things failed? would apply to all the panic(err) areas

Reply:
something like Error loading config or getting diffs or determining affected packages depending on the logic.

Reply:
Adding more descriptive error messages, also exiting with status 1 instead of panic for cleaner errors.